ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -31,6 +33,7 @@ interface PaywallDisplayCallback {
  */
 class PaywallActivityLauncher(resultCaller: ActivityResultCaller, resultHandler: PaywallResultHandler) {
     private val activityResultLauncher: ActivityResultLauncher<PaywallActivityArgs>
+    private val weakActivity = WeakReference(resultCaller as? Activity)
```

Comment:
Honestly, I believe this is quite hacky... But I wasn't able to figure out how to obtain a reference to the activity otherwise that would let us check if it was finishing from here... Open to ideas.